### PR TITLE
fix: pull-request.yml to ignore actions on ignore-sensitive-files-pr and Introspection workflow

### LINKED ISSUE #2383: Admin: Create PR GitHub Actions logic for the ignore-protected-files-pr label
**Is your feature request related to a problem? Please describe.**

1. The `pull-request.yml` workflow file has a section called `Check-Sensitive-Files`
1. This GitHub action:
    1. was created to flag PRs that edited sensitive files that we want to protect from unauthorized changes
    1. is run on all PRs except those submitted by `dependabot`
1. There are cases where community members need to edit sensitive files. When this is done, the `Check-Sensitive-Files` test fails. Our formal GitHub contributors can only merge PRs where all tests pass. This means the Repository owners have to merge the PRs. This creates a bottleneck in PR reviews

**Describe the solution you'd like**

1. There is a `ignore-sensitive-files-pr` label in this repository
1. We want only the `Check-Sensitive-Files` GitHub action to be ignored if this label is applied to the PR by any of our GitHub contributors
1. No other functionality must be altered, especially the logic where the  `Check-Sensitive-Files` GitHub action is also not run when the PR author is the `dependabot` user.

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- See above

**Additional context**
- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### LINKED ISSUE #2385: Fix the introspection GitHub action tests (Oct 2024)
**Is your feature request related to a problem? Please describe.**

1. Our introspection PR tests have been failing. This seems to be caused by code that references now deprecated Event and User task functionality.
2. Here is a recent example:
    1. https://github.com/PalisadoesFoundation/talawa-admin/actions/runs/11551343486/job/32148153150

**Describe the solution you'd like**

1. Update  the offending code to make the introspection tests pass
2. All existing tests must pass and must be valid tests of the code base
3. No other functionality must be affected

**Describe alternatives you've considered**

Here are some examples of issues fixing this in the past:

- https://github.com/PalisadoesFoundation/talawa-admin/issues/1542
- https://github.com/PalisadoesFoundation/talawa-admin/issues/1491

**Approach to be followed (optional)**

- See above

**Additional context**

- N/A

**Potential internship candidates**

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by govindsingh3477**
**What kind of change does this PR introduce?**

**Issue Number:**

Closes #2385 
Closes #2383 
**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

## Summary by CodeRabbit

- **New Features**
	- Introduced a consolidated `CREATE_CHAT` mutation for creating both group and direct chats with an additional `name` parameter.

- **Bug Fixes**
	- Enhanced GitHub Actions workflow to enforce stricter checks on sensitive files and target branches for pull requests.

- **Refactor**
	- Removed outdated `CREATE_GROUP_CHAT` and `CREATE_DIRECT_CHAT` mutations to streamline chat creation functionality.
	- Updated components to utilize the new `CREATE_CHAT` mutation instead of the removed direct chat mutation.

### CODE REVIEW

File: .github/workflows/pull-request.yml
Comment (varshith257) [inline .github/workflows/pull-request.yml:83]:
```suggestion
    if: ${{ github.actor != 'dependabot[bot]' && !contains(github.event.pull_request.labels.*.name, 'ignore-sensitive-files-pr)') }}
```

**Review by varshith257 — APPROVED** [review]

**Comment by varshith257** [comment]
@pranshugupta54 @AVtheking Can you approve the PR?

**Comment by Atharva-Kanherkar** [comment]
> @pranshugupta54 / @AVtheking / @Doraemon012 / @gautam-divyanshu / @Atharva-Kanherkar 
>   Can you approve the PR?

sure I'll review and tell you

**Review by gautam-divyanshu — APPROVED** [review]

**Review by Atharva-Kanherkar — APPROVED** [review]

**Comment by varshith257** [comment]
@palisadoes After 3 formal contributors' approvals. 
image

**Comment by varshith257** [comment]
@gautam-divyanshu @Atharva-Kanherkar Can you see the merge pull request button enabled on your screens? Can post the image?

**Comment by Atharva-Kanherkar** [comment]
> @gautam-divyanshu @Atharva-Kanherkar Can you see the merge pull request button enabled on your screens? Can post the image?

No I can not.

**Comment by varshith257** [comment]
@palisadoes I think there are restrictions enabled to branch

**Comment by varshith257** [comment]
Another option could be to give access to enable the auto-merge button if all workflow conditions are met and passes

**Review by Doraemon012 — APPROVED** [review]

**Comment by palisadoes** [comment]
1. I'm seeing it as mergeable.
2. Let me check the permissions.

**Comment by palisadoes** [comment]
@pranshugupta54 @gautam-divyanshu @AVtheking @varshith257

**Comment by palisadoes** [comment]
> it's still disabled image

Check your inbox. There may be an invitation to a group

**Comment by varshith257** [comment]
@palisadoes Still the same

**Comment by varshith257** [comment]
I am already in the new group

**Comment by palisadoes** [comment]
I've added the following persons to our new Maintainers Team.
 
1. @pranshugupta54
2. @AVtheking 
3. @Doraemon012 
4. @gautam-divyanshu 
5. @Atharva-Kanherkar 
6. @varshith257 

There are some factors to consider:

1. In PRs where sensitive files _`MUST`_ to be updated you can add the `ignore-sensitive-files-pr` label. This will allow you to merge the PR when all other tests pass.
2. In PRs where sensitive files _`MUST NOT`_ to be updated, do not add the label. Ask the PR's author to revert the changed file

**Pull request merged**
